Upstream: under review, https://github.com/cyrusimap/cyrus-sasl/pull/546

From 9c21fe64d69a9af08751c488a37d16b23e5a1cb0 Mon Sep 17 00:00:00 2001
From: Heiko Becker <heirecka@exherbo.org>
Date: Fri, 23 Nov 2018 17:02:51 +0100
Subject: [PATCH] Fix build with -Wl,--as-needed

Fails with "x86_64-pc-linux-gnu-ld: ../sasldb/.libs/libsasldb.a(db_gdbm.o):
in function `_sasldb_getdata': db_gdbm.c:(.text+0xcf): undefined
reference to `gdbm_open'..." otherwise.
---
 sasldb/Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sasldb/Makefile.am b/sasldb/Makefile.am
index 497ee25..a27645f 100644
--- a/sasldb/Makefile.am
+++ b/sasldb/Makefile.am
@@ -54,6 +54,6 @@ noinst_LTLIBRARIES = libsasldb.la
 
 libsasldb_la_SOURCES = allockey.c sasldb.h
 EXTRA_libsasldb_la_SOURCES = $(extra_common_sources)
-libsasldb_la_DEPENDENCIES = $(SASL_DB_BACKEND)
-libsasldb_la_LIBADD = $(SASL_DB_BACKEND)
+libsasldb_la_DEPENDENCIES = $(SASL_DB_BACKEND) $(SASL_DB_LIB)
+libsasldb_la_LIBADD = $(SASL_DB_BACKEND) $(SASL_DB_LIB)
 libsasldb_la_LDFLAGS = -no-undefined
-- 
2.19.2

